aiida_siesta/utils/converge_absclass.py

```diff
@@ -299,8 +299,29 @@ def _analyze_process(self, process_node):
 
             self.ctx.last_target_value = process_node.outputs.converged_target_value
 
+        else:
+            key = list(process_node.inputs.iterate_over.get_dict().keys())
+            self.report(
+                "\n WARINING: the next step of the sequential converger, if any, will be performed "
+                f"using the value of {key} specified in `converger_inputs`, not the last attempted value"
+            )
+
     def return_results(self):
 
         self.out('converged_parameters', DataFactory('dict')(dict=self.ctx.already_converged).store())
+
+        #Create and return a list of parameters that did not converged. We compare
+        #the iterate_over input and the self.ctx.already_converged. Both contains
+        #untraslated keywords.
+        unconv_list = []
+        iterate_over = self.ctx.inputs.iterate_over.get_list()
+        for pk_n in iterate_over:
+            step = load_node(pk_n).get_dict()
+            for par in step.keys():
+                if par not in self.ctx.already_converged:
+                    unconv_list.append(par)
```

Comment:
This will add the same key more than once in case it is unconverged more than once. 

You could check if the parameter is in `unconv_list` already. But I think it's better to first construct a list of all used parameters (possibly duplicated) and then just take the difference using a `set`.

```python
iterate_over = self.ctx.inputs.iterate_over.get_list()
# Get the list of all (possibly duplicated) parameters that have been iterated
all_parameters = [key for pk_n in iterate_over for key in load_node(pk_n).get_dict()]
# Find those that have not converged (while removing duplicates)
unconv_list = set(all_parameters).difference(self.ctx.already_converged)
```

Also it would be less nested (in my opinion easier to read).

PS: You would need to `list(unconv_list)` so that aiida is happy when you return the output.